### PR TITLE
Feat/#309 access control

### PR DESCRIPTION
**Pull request by ciaranschutte**
**Description of changes**

adds access control + tests
anything imported from https://github.com/icgc-argo/ego-token-utils/ not in `index` breaks

**Type of Change**

- [ ] Bug
- [ ] Styling
- [x] New Feature

**Checklist before requesting review:**

- [ ] Matches design:

  - component sizes, spacing, and styles
  - font size, weight, colour
  - spelling has been double checked

- [ ] New uikit components have Storybook stories
- [ ] Feature is minimally responsive.
- [x] Manual testing of UI feature.
- [ ] Add copyrights for new files
- [ ] Selenium test is completed and passing.

### CODE REVIEW

**Comment by ciaranschutte** [comment]
@hlminh2000 not sure how to fix the rollup config in https://github.com/icgc-argo/ego-token-utils/ to export from common.ts also.
Let me know if you think I should just put all constants we need into `index.ts` in https://github.com/icgc-argo/ego-token-utils/

**Comment by hlminh2000** [comment]
> @hlminh2000 not sure how to fix the rollup config in https://github.com/icgc-argo/ego-token-utils/ to export from common.ts also.
> Let me know if you think I should just put all constants we need into `index.ts` in https://github.com/icgc-argo/ego-token-utils/

we can do `export { ... } from 'common'` in `index.ts` eh, that'd be the best I think

File: src/routes/file-storage-api/accessValidations.test.ts [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+    });
+
+    it('allows access for programs associated with file', () => {
+      const file = {
+        ...baseFile,
+        ...{ release_stage: FILE_RELEASE_STAGE.OWN_PROGRAM },
+      };
+      const res = hasSufficientProgramMembershipAccess({ scopes, file });
+      expect(res).toBe(true);
+    }),
```
Comment (hlminh2000) [inline src/routes/file-storage-api/accessValidations.test.ts:127]: any purpose behind these commas?

File: src/routes/file-storage-api/accessValidations.ts [hunk 2]
```diff
@@ -1,28 +1,94 @@
...
 };
 
-export const hasSufficientDacoAccess = async (config: {
-  scopes: PermissionScopeObj[];
-  file?: EsFileCentricDocument;
-}): Promise<boolean> => {
-  return true;
+export const hasSufficientDacoAccess = (config: { scopes: PermissionScopeObj[] }): boolean => {
+  const dacoScope = config.scopes.find(({ policy }) => policy === 'DACO');
+  return dacoScope && dacoScope.permission === PERMISSIONS.READ ? true : false;
```
Comment (hlminh2000) [inline src/routes/file-storage-api/accessValidations.ts:91]: need to also check for `dacoScope.permission === PERMISSIONS.DENY` case

File: src/routes/file-storage-api/accessValidations.ts [hunk 3]
```diff
@@ -1,28 +1,98 @@
...
-  scopes: PermissionScopeObj[];
-  file?: EsFileCentricDocument;
-}): Promise<boolean> => {
-  return true;
+export const hasSufficientDacoAccess = (config: { scopes: PermissionScopeObj[] }): boolean => {
+  const dacoScopes = config.scopes.filter(({ policy }) => policy === EGO_DACO_POLICY_NAME);
+
+  return dacoScopes.length > 0 && dacoScopes.every(scope => scope.permission === PERMISSIONS.READ)
+    ? true
+    : false;
```
Comment (hlminh2000) [inline src/routes/file-storage-api/accessValidations.ts:95]: `dacoScopes.length > 0 && dacoScopes.every(scope => scope.permission === PERMISSIONS.READ)` is already a positive boolean, no need for this extra boolean conversion :)

**Review by hlminh2000 — APPROVED** [review]
tiny improvement :)

**Pull request merged**